src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,54 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Bulldozer;
+import core.mate.academy.model.Excavator;
+import core.mate.academy.model.Machine;
+import core.mate.academy.model.Truck;
+import core.mate.academy.service.producer.BulldozerProducer;
+import core.mate.academy.service.producer.ExcavatorProducer;
+import core.mate.academy.service.producer.MachineProducer;
+import core.mate.academy.service.producer.TruckProducer;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+    private static final MachineProducer<Bulldozer> bulldozerProducer = new BulldozerProducer();
+    private static final MachineProducer<Excavator> excavatorProducer = new ExcavatorProducer();
+    private static final MachineProducer<Truck> truckProducer = new TruckProducer();
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machines = new ArrayList<>();
+
+        if (type == Bulldozer.class) {
+            machines = bulldozerProducer.get();
+        }
+
+        if (type == Excavator.class) {
+            machines = excavatorProducer.get();
+        }
+
+        if (type == Truck.class) {
+            machines = truckProducer.get();
+        }
+
+        return new ArrayList<>(machines);
```

Comment:
Returning a new ArrayList<>(machines) is unnecessary if no producer has been found. Instead, return Collections.emptyList() to avoid unnecessary object creation.